.github/workflows/test-e2e-with-modulereleasemeta.yml

```diff
@@ -0,0 +1,103 @@
+name: TestSuite E2E with ModuleReleaseMeta
```

Comment:
I think it's too early that you introduce a new setup with ModuleReleaseMeta, and in this setup, you have the test case that have `ModuleReleaseMeta` and old naming Moduletemplate both exists, right? What surprised me is that all the e2e tests passed. But according to your new [logic](https://github.com/kyma-project/lifecycle-manager/pull/1947/files#diff-13e8e3c3d462afb13b63d6f7c09cb4036aa694e28fb3df9c1ba6ae21d40ddd29R100), if `ModuleReleaseMeta` exists, it fetches ModuleTemplate with new naming (modulename+version), and since you are just [apply](https://github.com/kyma-project/lifecycle-manager/pull/1947/files#diff-328e54a8869347e6d4401cafdc31b8d28331fd651340f2ec5a954880b58e0897R64-R66) old naming Moduletemplate, the test should failed, right? or am I missing something here.

But anyway, I suggest we should not introduce this new setup for all existing tests, basically, we have 3 stages:
1. only old name Moduletemplate
2. old Moduletemplate + ModuleReleaseMeta (migration period)
3. new Moduletemplate + ModuleReleaseMeta

In your PR, we just need to introduce two test cases for 2, 3, and the existing e2es should covered case 1.

And before we fully migrated to stage 3, we start to adapt all existing e2es tests.

Reply:
The name of the ModuleTemplate is being overwritten in the e2e with ModuleReleaseMeta workflow, and I have a ticket created to remove the overwrite when the modulectl supports this.

As agreed, I will add a test case to support number 2 (ensuring the Kyma will have the expected error when there is an old ModuleTemplate with ModuleReleaseMeta)